alg/teca_python_vertical_reduction.py

```diff
@@ -62,6 +64,18 @@ def request(self, port, input_md, request):
         # transform extent, add back the vertical dimension
         md = teca_metadata(input_md[0])
 
+        # get the device to run on
+        dev = -1
+        np = numpy
+        if get_teca_has_cuda() and get_teca_has_cupy():
+            dev = request['device_id']
+            if dev >= 0:
+                cupy.cuda.Device(dev).use()
+                np = cupy
```

Comment:
The metadata is CPU based. Unless we're doing a heavy calculation, probably best to use CPU in the report and request phase.